specs/WebNotification.md

```diff
@@ -0,0 +1,747 @@
+            message +=
+                (userInitiated ? L"This request came from a user gesture."
+                               : L"This request did not come from a user gesture.");
+
+            int response = MessageBox(
+                nullptr, message.c_str(), L"Permission Request",
+                MB_YESNOCANCEL | MB_ICONWARNING);
+            switch (response)
+            {
+            case IDYES:
+                m_cachedPermissions[cachedKey] = true;
+                state = COREWEBVIEW2_PERMISSION_STATE_ALLOW;
+                break;
+            case IDNO:
+                m_cachedPermissions[cachedKey] = false;
+                state = COREWEBVIEW2_PERMISSION_STATE_DENY;
+                break;
+            default:
+                state = COREWEBVIEW2_PERMISSION_STATE_DEFAULT;
+                break;
+            }
+        }
+
+        CHECK_FAILURE(args->put_State(state));
+        CHECK_FAILURE(deferral->Complete());
+    });
+    return S_OK;
+}
+
+```
+
+## Filter Notifications from a specific doamin and send local toast
+
+This sample uses custom app UI to display toast notifications. It does so using
+`Microsoft.Toolkit.Uwp.Notifications` nuget package. For an unpackaged C# WinApp
+SDK based app, you can handle the toast notification activation in your app's
+startup code such as in `App.xaml.cs`, and then dispatch it to the WebView2
+notification object. Note that the notification object will be gone along with
+WebView2. If your app is closed when the toast notification is activated, the
+notification object should be gone and you should not try to dispatch the
+activation.
+
+For UWP and desktop packaged apps, you can learn more about via [Send a local
+toast notification from a C# app - Windows apps | Microsoft
+Learn](https://learn.microsoft.com/windows/apps/design/shell/tiles-and-notifications/send-local-toast?tabs=uwp).
+
+### C# Sample
+
+```csharp
+using Microsoft.Toolkit.Uwp.Notifications;
+...
+void WebView_NotificationReceived(object sender, CoreWebView2NotificationReceivedEventArgs args)
+{
+    using (args.GetDeferral())
+    {
+    args.Handled = true;
+    var notification = args.Notification;
+
+    // Show notification with local MessageBox
+    System.Threading.SynchronizationContext.Current.Post((_) =>
+    {
+        string message = "Received notification from " + args.Uri + " with body " + notification.Body;
+        MessageBox.Show(message);
+    }
+
+    // Requires Microsoft.Toolkit.Uwp.Notifications NuGet package version 7.0 or greater
+    var toastContent = new ToastContentBuilder()
+        .AddArgument("conversationId", m_id)
+        .AddText("Notification sent from " + args.Uri +":\n")
+        .AddText(notification.Body);
+    toastContent.Show();
+
+    // Call ReportShown after showing the toast notification to raise
+    // the DOM notification.show event.
+    // args.Handled has been set to true before we call Report... methods.
+    notification.ReportShown();
```

Comment:
Why is ReportShown() void but the other Report* functions return IAsyncAction ?  The sample seems to ignore the returned values in all cases.